### ISSUE TITLE
Does this still work?

### ISSUE THREAD

**Issue by agentdouble0** [benotter/OVRLay#3]
I know it hasn't been updated in years but it seems pretty well done and semi-simple to use. But I am trying to make a program similar to OVRDrop/OVR Toolkit and what not. But the issue I am having is I have no idea how to take what I've made for an overlay (the desktop dup and UI) and actually make it into an overlay in VR once the program is running.

I found a few different GitHub projects that help with making it but they are so outdated and using really old versions of unity that it's near impossible to use. I found this one that isn't ungodly outdated and using a NEWER unity so I figure ill give it a shot.

I am not sure if it works (it might and it kind still works with a few tweaks) and if it does how to really use it to get an overlay in VR.

I followed the steps of the Moderate Usage and it certainly does something, but I am not sure what it should be doing and exporting what I've created and run it certainly does not give me an overlay in VR.

Not sure if you will respond to this as this is a very old project but I figure I would give it shot.

Just hoping you can confirm if this is still kind of working/can be used to make a working VR Overlay and maybe helping my brain dead ass figure out how to make a VR Overlay in general.

(I am using/attempting to use **Unity 2019.3.0f6** with this project btw)

**Comment by ryuyandev** [comment]
I was able to get this work in 2018.4.20f1 (haven't tried newer versions yet). But this relies on inspecting running processes to see if VR is running, so you will either have to run your application as administrator, or add your application to your steam library (click add a game in the bottom left corner of the library and then click add a non-steam game). See if that helps.

**Comment by agentdouble0** [comment]
> I was able to get this work in 2018.4.20f1 (haven't tried newer versions yet). But this relies on inspecting running processes to see if VR is running, so you will either have to run your application as administrator, or add your application to your steam library (click add a game in the bottom left corner of the library and then click add a non-steam game). See if that helps.

Thank you for responding and I am very surprised that someone responded, to be honest! It's been a while since I visited my project of mine so I'll have to refresh myself on what I am doing and what I need to do. 

I do believe I ran it in Administrative mode but not 100% sure but I'll try it again or just add it to the steam library as you said.

I will continue to try to get this to work on my version of Unity as I think it should work (as it does seem to work for the most part) if not I will attempt to downgrade and see what happens.

I'll let you know if I can get it working and again thank you for responding!

**Comment by agentdouble0** [comment]
> I was able to get this work in 2018.4.20f1 (haven't tried newer versions yet). But this relies on inspecting running processes to see if VR is running, so you will either have to run your application as administrator, or add your application to your steam library (click add a game in the bottom left corner of the library and then click add a non-steam game). See if that helps.


I did finally get it to work so thanks for that. With that said it seems to be a little bit buggy (both on my end of code and the code from this) so I will have to figure that out. But for now, I can continue my work. I am just a little mad that it was just a simple fix that I just didn't think to try *-.-

Thanks again!

**Comment by ryuyandev** [comment]
Oh, wow. It had been a while since you posted the issue so I'd worried you may have moved on. I'm really glad to hear that I was able to help!

Just FYI, while you should be able to create basic stuff just fine with this package, I warn you that it appears to be using its own included version of the openvr api which appears to be outdated. You can't use this with the official steamvr plugin, should your project need something from there. I'll likely fork this and make my own version that includes the steamvr plugin as a dependency.

Good luck with your project!

**Comment by agentdouble0** [comment]
> Oh, wow. It had been a while since you posted the issue so I'd worried you may have moved on. I'm really glad to hear that I was able to help!
> 
> Just FYI, while you should be able to create basic stuff just fine with this package, I warn you that it appears to be using its own included version of the openvr api which appears to be outdated. You can't use this with the official steamvr plugin, should your project need something from there. I'll likely fork this and make my own version that includes the steamvr plugin as a dependency.
> 
> Good luck with your project!

Yeah it is very outdated so somethings that I am trying to use it for doesn't work too well and it's missing features here and there, if you were to make your own updated version that would be great! Let me know if you do!

I was able to create a proper overlay and a dashboard menu but the image doesn't load for some reason no matter what I put in for the texture. Everything else seems to be working. 

As for my project, I am very close to finishing it (all logic is working, menus are working, etc) but the main issue I have to figure out is how to make my duplicated screen (using Windows API) interactable in VR as, as of now, all it does is duplicated the screen selected (if you have multiple) but you can't it control anything in VR.

Close but not quite there.
